androidApp/src/main/java/dev/johnoreilly/confetti/sessions/navigation/SessionsDestination.kt

```diff
@@ -30,22 +43,41 @@ fun NavGraphBuilder.sessionsGraph(
     navigateToSignIn: () -> Unit,
     onSignOut: () -> Unit,
     onSwitchConferenceSelected: () -> Unit,
+    defaultConferenceParameter: String?,
 ) {
     composable(
         route = sessionsRoutePattern,
         arguments = listOf(
             navArgument(conferenceArg) {
                 type = NavType.StringType
+                nullable = true
+                defaultValue = defaultConferenceParameter
+            }
+        ),
+        deepLinks = listOf(
+            navDeepLink {
+                uriPattern = "confetti://confetti/sessions/{$conferenceArg}"
             }
         )
-    ) { backStackEntry ->
-        SessionsRoute(
-            conference = SessionsKey(backStackEntry).conference,
-            appState = appState,
-            navigateToSession = navigateToSession,
-            onSignOut = onSignOut,
-            navigateToSignIn = navigateToSignIn,
-            onSwitchConferenceSelected = onSwitchConferenceSelected,
-        )
+    ) {
+        val homeViewModel: HomeViewModel = koinViewModel()
+        when (val homeUiState = homeViewModel.uiState.collectAsStateWithLifecycle().value) {
+            HomeUiState.Loading -> {}
+            HomeUiState.NoConference -> {
+                LaunchedEffect(Unit) {
+                    onSwitchConferenceSelected()
+                }
+            }
+            is HomeUiState.Conference -> {
+                SessionsRoute(
+                    conference = homeUiState.conference,
+                    appState = appState,
+                    navigateToSession = navigateToSession,
+                    onSignOut = onSignOut,
+                    navigateToSignIn = navigateToSignIn,
+                    onSwitchConferenceSelected = onSwitchConferenceSelected,
+                )
+            }
+        }
     }
```

Comment:
This "extra layer" sort of was added in the sessions destination to decide on if it show the sessions route or to go to the conferences destination, without having to add a third helper destination to support this.